include/maidsafe/common/rsa.h

```diff
@@ -85,10 +86,15 @@ PlainText Decrypt(const CipherText& data, const PrivateKey& private_key);
 
 Signature Sign(const PlainText& data, const PrivateKey& private_key);
 
+std::vector<byte> Sign(const std::vector<byte>& data, const PrivateKey& private_key);
```

Comment:
Yes we could also look at bounded vector types as well I suppose (see David Stones bounded integer library) with bounded sizes. I suppose if the type here is an RsaSignature<T> then it may make even more flexible, with aliases for array and string if we wished. Seems like getting deeper, but may be correct when we have multiple representations of the same thing? Or as you say for now maybe sticking to Signature is the best, certainly the fastest and I suppose we could do some retrofitting with fundamental (templated) types later on.